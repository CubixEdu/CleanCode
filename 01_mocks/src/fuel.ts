export class Fuel {

    public fill(litersOfFuel: number) {
        return litersOfFuel;
    }
}