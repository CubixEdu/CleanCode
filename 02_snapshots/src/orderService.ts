import { UserService } from "./userService";

export class OrderService {

    constructor(private userService: UserService) { }

    public getUsersOfTheLastHour() {
        const users = [];
        for (let index = 1; index < 5; index++) {
            const user = this.userService.getUserById(index);
            users.push(user);
        }
        return users;
    }
}