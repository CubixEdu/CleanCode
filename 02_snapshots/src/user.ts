export class User {
  
    constructor(
      private id: number,
      private username: string,
      private email: string,
      private firstName: string,
      private lastName: string,
      private dateOfBirth: Date,
      private address: {
        street: string;
        city: string;
        zipCode: string;
      },
      private phoneNumber: string,
      private isActive: boolean,
      private roles: string[]
    ) {}
  }