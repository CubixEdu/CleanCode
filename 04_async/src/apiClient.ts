import { delay } from "./common";

export type Response = {
    data: string;
    statusCode: number;
}

export class ApiClient {

    public async downloadUsernames() {
        await delay(2000);
        return ["batman", "superman", "wonderwoman"];
    }

    public async fetchFromUrl(apiUrl: string) : Promise<Response> {
        await delay(100);
        return { data: "some data", statusCode: 200 };
    }
}