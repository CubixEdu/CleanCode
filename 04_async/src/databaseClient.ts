import { delay } from "./common";

export class DatabaseClient {

    public async getAllUsernames() : Promise<string[]> {
        await delay(3000);
        return ["batman", "superman", "wonderwoman"];
    }

    public async saveIntoDatabase(usernames: string[]): Promise<void> {
        await delay(1500);
    }
}