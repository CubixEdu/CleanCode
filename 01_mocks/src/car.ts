import { Fuel } from "./fuel";

export class Car {

    constructor(private fuel: Fuel) { }

    public drive() {
        this.fuel.fill(35);
    }
}