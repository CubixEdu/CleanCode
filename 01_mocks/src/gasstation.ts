import { Fuel } from "./fuel";

export class GasStation {

    constructor(private fuel: Fuel) { }

    public fillUpCars(numberOfCars: number) {
        for (let index = 1; index < numberOfCars + 1; index++) {
            this.fuel.fill(index + 1);
        }
    }
}