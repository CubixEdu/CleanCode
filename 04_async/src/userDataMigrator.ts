import { ApiClient } from "./apiClient";
import { DatabaseClient } from "./databaseClient";

export class UserDataMigrator {

    constructor(private dbClient: DatabaseClient, private apiClient: ApiClient) {}

    public async migrate() {
        const usernamesFromApi = await this.downloadFromApi();
        await this.dbClient.saveIntoDatabase(usernamesFromApi);
    }

    public async canDownload(apiUrl: string) {
        const response = await this.apiClient.fetchFromUrl(apiUrl);
        if (response.statusCode === 200) {
            return true;
        }

        return false;
    }

    private async downloadFromApi() {
        return await this.apiClient.downloadUsernames();
    }

    private async saveIntoDatabase(usernames: string[]) {
        await this.dbClient.saveIntoDatabase(usernames);
    }
}