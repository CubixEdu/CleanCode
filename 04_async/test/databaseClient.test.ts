import { DatabaseClient } from "../src/databaseClient"

describe('DatabaseClient tests', () => {
    it('should return all the usernames', async () => {
        // Arrange
        const dbClient = new DatabaseClient();

        // Act
        const result = await dbClient.getAllUsernames();

        // Assert
        expect(result).toMatchSnapshot();

    })
})
