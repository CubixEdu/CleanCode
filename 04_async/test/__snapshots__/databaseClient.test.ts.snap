// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatabaseClient tests should return all the usernames 1`] = `
[
  "batman",
  "superman",
  "wonderwoman",
]
`;
